dronecaster.lua

```diff
@@ -57,6 +66,71 @@ done_init = false
 
 -- init & core
 --------------------------------------------------------------------------------
+
+-- midi
+function build_midi_device_list()
+  midi_devices = {}
+  for i = 1,#midi.vports do
+    local long_name = midi.vports[i].name
+    local short_name = string.len(long_name) > 15 and util.acronym(long_name) or long_name
+    table.insert(midi_devices,i..": "..short_name)
+  end
+end
+
+function midi_event(data)
+  -- global midi setting
+  if midi_enabled == 1 then
+    msg = midi.to_msg(data)
+    -- filter channel
+    if msg.ch == midi_channel then
+      if msg.type ~= "clock" then
+        tab.print(msg)
+      end
+      --note message
+      if msg.type == "note_on" then
+        params:set("note",msg.note)
+        -- amp control
+        print(midi_amp_control)
+        if midi_amp_control == 2 then
+          amp = math.min(msg.vel/100, 1)
+        end
+        if midi_amp_control == 3 then
+          amp = math.min(msg.key_pressure/100, 1)
```

Comment:
This could be fun 😃

Reply:
Er wait, I think I misread this too. I was thinking of channel pressure or aftertouch, but it looks like this is within the note handling block. I'm not even sure what note key pressure is! 🤔

Reply:
Hah I might be confused too, I want this to be aftertouch but channel pressure does not come along with note messages. I don't actually have a real keyboard with aftertouch to test with though.